rep-I0005.rst

```diff
@@ -0,0 +1,164 @@
+::
+
+   REP: I0005
+   Title: Cartesian Motion Extensions for the Simple Message Protocol
+   Authord: Fred Proctor <proctor@nist.gov> and Tre Harrison <william.harrison@nist.gov>
+   Status: Active
+   Type: Process
+   Content-Type: text/x-rst
+   Created: 20-May-2015
+   Post-History: 20-May-2015
+
+Outline
+=======
+
+#. Abstract_
+#. Motivation_
+#. `Message Specifications`_
+#. `Cartesian Move Request`_
+#. `Cartesian Move Reply`_
+#. `Cartesian Move State`_
+#. `Object State`_
+#. References_
+#. `Revision History`_
+#. Copyright_
+
+Abstract
+========
+
+This REP documents extensions to the Simple Message protocol
+that support Cartesian motion, and the reporting of information about
+objects in the robot's workspace.
+
+Motivation
+==========
+
+The Simple Mesage protocol defines a low-level interface to robot
+controllers that can easily be implemented. Together with reference
+implementations of Robot Operating System (ROS) trajectory planners,
+Simple Message provides a way for robot vendors to quickly provide
+baseline ROS functionality to their systems. 
+
+Simple Message is a joint-level interface that requires the external
+system to perform the inverse kinematics calculations. Robot
+controllers typically provide this function internally, and the
+proposed extensions enable external systems to benefit from the
+vendor's built-in kinematics. 
+
+The extensions also include messages for reporting the location and
+other information about objects in the robot's immediate environment
+that may be tracked using built-in capabilities of the robot
+controller. 
+
+Message Specifications
+======================
+
+Cartesian Move Request
+----------------------
+A Cartesian move message requests that the robot move to the specified
+Cartesian position and quaternion orientation along a straight line
+and about a fixed rotational axis. The translational speed applies to
+the motion along the straight line. The rotational speed applies to
+the motion about the fixed axis. Translational and rotational motion
+should finish at the same time, and in general one of these speeds may
+be slower than specified to ensure simultaneous arrival. The sequence
+number provided will be used in subsequent messages from the robot
+controller to associate replies and status with the move request.
+
+Format of a Cartesian move request::
+
+  int length: 4 bytes, constant value 13 x 4 = 52
+  int message_type: 4 bytes, constant value 31
+  int comm_type; 4 bytes, constant value 2, REQUEST
+  int reply_type: 4 bytes, ignored, any value, typically 0
+  int seq_number: 4 bytes, identifying this request
+  float x, y, z, qx, qy, qz, qw: 4 bytes each, for the requested
+  Cartesian position and quaternion rotation
+  float translational_speed: 4 bytes, the speed to be applied to the
+  translational part
+  float rotational_speed: 4 bytes, the speed to be applied to the
+  rotational part
```

Comment:
While it hasn't happened yet, the `shared...` types were meant to support controllers with different variable sizes (since it was very unlikely that the controller would allow you to specify the size).  However, for the purposes of standardization, it is helpful to know the "standard" size.